### ISSUE TITLE
ESlint isn't showing info about type errors

### ISSUE THREAD

**Issue by onhernandes** [typescript-eslint/typescript-eslint#2254]
I don't know if I'm expecting the wrong thing here but, I want to have errors messages from eslint about type checking, but isn't working - I made the whole setup through `eslint --init`, I added manually the `standard-with-typescript` plugin.

**Repro**

```JSON
{
    "env": {
        "es6": true,
        "node": true
    },
    "extends": [
        "standard-with-typescript"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "project": "./tsconfig.json",
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
      "@typescript-eslint/no-require-imports": 2
    }
}

```

My TS config:

```JSON
{
   "compilerOptions": {
      "lib": [
         "es5",
         "es6"
      ],
      "target": "es2019",
      "module": "commonjs",
      "moduleResolution": "node",
      "outDir": "./build",
      "emitDecoratorMetadata": true,
      "experimentalDecorators": true,
      "sourceMap": true,
      "esModuleInterop": true
   },
   "files": ["test.ts"]
}
```

```TS
import Koa from 'koa'

const test = (): string => {
  return new Koa()
}

test()
```

**Expected Result**
ESlint should emit error messages about the type error (returning a Koa App instead string)

**Actual Result**
It works and shows errors based on my TS config, like when I try to eslint another file that's not `test.ts`, and it shows the error for `no-require-imports`, but I isn't showing errors about my types.

**Additional Info**

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `3.4.0` |
| `TypeScript`                | `3.9.5` |
| `ESLint`                    | `7.3.1` |
| `node`                      | `12.16.2` |
| `npm`                       | `6.14.4` |

**Comment by bradzacher** [comment]
Out tooling does not show typescript errors.

See #2211

**Comment by onhernandes** [comment]
Oh, I didn't know that. Thanks.
